crates/interpreter/src/instructions/opcode.rs

```diff
@@ -1,453 +1,313 @@
-pub const PUSH15: u8 = 0x6e;
-pub const PUSH16: u8 = 0x6f;
-pub const PUSH17: u8 = 0x70;
-pub const PUSH18: u8 = 0x71;
-pub const PUSH19: u8 = 0x72;
-pub const PUSH20: u8 = 0x73;
-pub const PUSH21: u8 = 0x74;
-pub const PUSH22: u8 = 0x75;
-pub const PUSH23: u8 = 0x76;
-pub const PUSH24: u8 = 0x77;
-pub const PUSH25: u8 = 0x78;
-pub const PUSH26: u8 = 0x79;
-pub const PUSH27: u8 = 0x7a;
-pub const PUSH28: u8 = 0x7b;
-pub const PUSH29: u8 = 0x7c;
-pub const PUSH30: u8 = 0x7d;
-pub const PUSH31: u8 = 0x7e;
-pub const PUSH32: u8 = 0x7f;
-pub const DUP1: u8 = 0x80;
-pub const DUP2: u8 = 0x81;
-pub const DUP3: u8 = 0x82;
-pub const DUP4: u8 = 0x83;
-pub const DUP5: u8 = 0x84;
-pub const DUP6: u8 = 0x85;
-pub const DUP7: u8 = 0x86;
-pub const DUP8: u8 = 0x87;
-pub const DUP9: u8 = 0x88;
-pub const DUP10: u8 = 0x89;
-pub const DUP11: u8 = 0x8a;
-pub const DUP12: u8 = 0x8b;
-pub const DUP13: u8 = 0x8c;
-pub const DUP14: u8 = 0x8d;
-pub const DUP15: u8 = 0x8e;
-pub const DUP16: u8 = 0x8f;
-pub const SWAP1: u8 = 0x90;
-pub const SWAP2: u8 = 0x91;
-pub const SWAP3: u8 = 0x92;
-pub const SWAP4: u8 = 0x93;
-pub const SWAP5: u8 = 0x94;
-pub const SWAP6: u8 = 0x95;
-pub const SWAP7: u8 = 0x96;
-pub const SWAP8: u8 = 0x97;
-pub const SWAP9: u8 = 0x98;
-pub const SWAP10: u8 = 0x99;
-pub const SWAP11: u8 = 0x9a;
-pub const SWAP12: u8 = 0x9b;
-pub const SWAP13: u8 = 0x9c;
-pub const SWAP14: u8 = 0x9d;
-pub const SWAP15: u8 = 0x9e;
-pub const SWAP16: u8 = 0x9f;
-pub const RETURN: u8 = 0xf3;
-pub const REVERT: u8 = 0xfd;
-pub const INVALID: u8 = 0xfe;
-pub const ADDRESS: u8 = 0x30;
-pub const BALANCE: u8 = 0x31;
-pub const BASEFEE: u8 = 0x48;
-pub const ORIGIN: u8 = 0x32;
-pub const CALLER: u8 = 0x33;
-pub const CALLVALUE: u8 = 0x34;
-pub const GASPRICE: u8 = 0x3a;
-pub const EXTCODESIZE: u8 = 0x3b;
-pub const EXTCODECOPY: u8 = 0x3c;
-pub const EXTCODEHASH: u8 = 0x3f;
-pub const RETURNDATASIZE: u8 = 0x3d;
-pub const RETURNDATACOPY: u8 = 0x3e;
-pub const BLOCKHASH: u8 = 0x40;
-pub const COINBASE: u8 = 0x41;
-pub const TIMESTAMP: u8 = 0x42;
-pub const NUMBER: u8 = 0x43;
-pub const DIFFICULTY: u8 = 0x44;
-pub const GASLIMIT: u8 = 0x45;
-pub const SELFBALANCE: u8 = 0x47;
-pub const SLOAD: u8 = 0x54;
-pub const SSTORE: u8 = 0x55;
-pub const GAS: u8 = 0x5a;
-
-pub const LOG0: u8 = 0xa0;
-pub const LOG1: u8 = 0xa1;
-pub const LOG2: u8 = 0xa2;
-pub const LOG3: u8 = 0xa3;
-pub const LOG4: u8 = 0xa4;
-
-pub const CREATE: u8 = 0xf0;
-pub const CREATE2: u8 = 0xf5;
-pub const CALL: u8 = 0xf1;
-pub const CALLCODE: u8 = 0xf2;
-pub const DELEGATECALL: u8 = 0xf4;
-pub const STATICCALL: u8 = 0xfa;
-pub const SELFDESTRUCT: u8 = 0xff;
-pub const CHAINID: u8 = 0x46;
+impl fmt::Display for OpCode {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        let n = self.get();
+        if let Some(val) = OPCODE_JUMPMAP[n as usize] {
+            f.write_str(val)
+        } else {
+            write!(f, "UNKNOWN(0x{n:02X})")
+        }
+    }
+}
 
 impl OpCode {
```

Comment:
Deprecated:
- `try_from_u8` -> `new`
- `u8` -> `get`

This is the same naming scheme as `NonZero*` std numbers.